### PR TITLE
Migrated from unittest to pytest

### PR DESCRIPTION
**Pull request by Alien1993**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,13 +6,15 @@ python: "2.7"
 env:
```
Comment (palazzem) [inline .travis.yml]: I saw that you're extending the python support in the ``tox.ini`` file, but you're not executing the new test suite here. Fill the ``TOX_ENV`` list.

File: runtest.py [hunk 2]
```diff
@@ -1,10 +1,4 @@
-import unittest
 import sys
+import pytest
```
Comment (palazzem) [inline runtest.py]: I usually add this [comment][1] before the ``sys.exit()`` call. Just as a reminder.

[1]: https://github.com/palazzem/wagtail-nesting-box/blob/master/runtests.py#L6

File: tests/compat.py [hunk 3]
```diff
@@ -1,4 +0,0 @@
-try:
```
Comment (palazzem) [inline tests/compat.py]: Removing this ``compat`` module is OK, but you can import ``mock`` just because ``pytest-mock`` requires the ``mock`` package that adds ``mock`` functionality to ``python2.7``.

I'm going to double check all tests where you're using ``import mock`` because it's better to use the ``mocker`` fixture instead of ``mock`` (you have installed ``pytest-mock`` for that).

Why bother using ``mocker`` instead of ``mock``? Check [pytest-mock][1] documentation ("*Why bother with a plugin?*" section)

[1]: https://pypi.python.org/pypi/pytest-mock/1.2

File: tests/tests_integration/test_api.py [hunk 4]
```diff
@@ -1,11 +1,13 @@
+from . import missing_app_tokens
 from mkmsdk.api import Api
-from . import IntegrationTest
 
 
-class ApiTest(IntegrationTest):
-    def setUp(self):
```
Comment (palazzem) [inline tests/tests_integration/test_api.py]: Here we can discuss a lot. What I'm saying applies to all tests and is just *an opinion* about how to write readable tests.

First, I usually avoid to create classes (in this case ``TestApi``) when using ``pytest`` because you really don't need them, unless you're grouping together different set of tests in the same file.

**And what about the ``setup_method``?**
I usually avoid it because it doesn't make clear what the test is doing. If you have a very long file and you're fixing a broken test, you have to read the test, go at the top of the file looking for what the ``setUp`` method does and then go back to the test trying to get what it *really* does.

So, if you have 4 tests that must use the ``Api(sandbox_mode=True)`` just declare it in each test. This prevents any side effect making the test more readable, because everything is needed in the test is in the test itself.

**And what about DRY?**
Simply use ``pytest`` [fixtures][1] and "inject" them in your tests.

To sum up, I suggest to:
* remove the class "wrapper" where it applies
* remove all ``setup_method``
* put what the ``setup_method`` does in each test
* if you think that it could be a reusable component, create a fixture

[1]: http://doc.pytest.org/en/latest/fixture.html

File: tests/tests_integration/test_mkm.py [hunk 5]
```diff
@@ -19,14 +19,16 @@ def test_response_is_as_expected(self):
...
 
-        first_game_received = json_response['game']
+        received_games = json_response['game']
 
-        self.assertEqual(first_game_received, expected_response['game'], 'Game received is not correct')
+        assert received_games == expected_response['game']
 
     def test_sandbox_url(self):
         response = mkm_sandbox.market_place.games()
-        self.assertEqual(response.request.url, 'https://sandbox.mkmapi.eu/ws/v1.1/output.json/games')
```
Comment (palazzem) [inline tests/tests_integration/test_mkm.py]: I think it's reasonable in the integration tests to assert the ``status_code == 200``. This enforces the response check and helps you to quickly understand why the test didn't pass. For instance, it's good to know that your integration test in your CI is not passing because the API answered with an ``Internal Server Error (500)`` and not because the ``request.url`` is not defined.

Anyway, check the expected ``status_code`` where it is applicable.

File: tests/tests_integration/test_oauth.py [hunk 6]
```diff
@@ -70,31 +67,12 @@ def test_account_entity_is_as_expected(self):
...
 
         received_first_name = json_response['account']['name']['firstName']
         received_last_name = json_response['account']['name']['lastName']
 
-        expected_first_name = expected_account_response['account']['name']['firstName']
-        expected_last_name = expected_account_response['account']['name']['lastName']
-
-        self.assertEqual(received_first_name, expected_first_name, 'First name received is not as expected')
-
-        self.assertEqual(received_last_name, expected_last_name, 'Last name received is not as expected')
```
Comment (palazzem) [inline tests/tests_integration/test_oauth.py]: This test is considerably different. Is it what you're looking for? Furthermore ``received_first_name`` is a string right? if it's a string, just use ``assert received_first_name`` without ``len()``. Few examples:
```
>>> a = ""
>>> assert a
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError
>>> a = None
>>> assert a
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError
>>> a = "a"
>>> assert a
# that's OK
```

File: tests/tests_unit/test_api.py [hunk 7]
```diff
@@ -36,96 +34,97 @@ def test_create_auth_with_empty_string_token(self):
...
-        self.assertEqual(auth.client.client_secret, 'app_secret')
-        self.assertEqual(auth.client.resource_owner_key, '')
-        self.assertEqual(auth.client.resource_owner_secret, '')
+        assert isinstance(auth.client, MKMClient)
+        assert auth.client.client_key == 'app_token'
+        assert auth.client.client_secret == 'app_secret'
+        assert auth.client.resource_owner_key == ''
+        assert auth.client.resource_owner_secret == ''
 
     def test_missing_env_var_raise_exception_correctly(self):
```
Comment (palazzem) [inline tests/tests_unit/test_api.py]: ``mocker`` is your friend:
```python
os_mocked = mocker.patch('mkmsdk.os')

File: tests/tests_unit/test_requests.py [hunk 8]
```diff
@@ -1,14 +1,12 @@
...
 
-class RequestsTest(unittest.TestCase):
-
+class TestRequests:
     def test_request_works(self):
         url = '/metaproducts/island/1/1'
         method = 'get'
 
         with mock.patch('mkmsdk.api.Api.request') as mock_request:
             mock_request(url, method)
```
Comment (palazzem) [inline tests/tests_unit/test_requests.py]: This test is significantly different (before you were outside the context manager). Is it OK?

File: tox.ini [hunk 9]
```diff
@@ -1,36 +1,28 @@
 [tox]
 envlist =
-    py27, py34,
-    docs, flake8
```
Comment (palazzem) [inline tox.ini]: It's great that you're supporting so many python versions! Write that into the ``README``

**Comment by Alien1993** [comment]
Fixes #5

**Comment by Alien1993** [comment]
Fixes #12
